ldpd/labelmapping.c

```diff
@@ -723,6 +723,14 @@ tlv_decode_fec_elm(struct nbr *nbr, struct ldp_msg *msg, char *buf,
 		/* Prefix Length */
 		map->fec.prefix.prefixlen = buf[off];
 		off += sizeof(uint8_t);
+		if ((map->fec.prefix.af == AF_IPV4
+		     && map->fec.prefix.prefixlen > IPV4_MAX_PREFIXLEN)
```

Comment:
should we check for != rather than '>' ? isn't receiving fec prefix with lesser length also wrong ?

Reply:
No. RFC 5036 says the following (section 2.1):
```
   This specification defines a single type of FEC element, the "Address
   Prefix FEC element".  This element is an address prefix of any length
   from 0 to a full address, inclusive.
```